### PR TITLE
feat: add `dob` sub-package for spore-sdk

### PR DESCRIPTION
**Pull request by ashuralyk**
# Description
since we created `DOB/0` protocol, this protocol requires developer write RUST code to generate its `pattern` for deploying Cluster, so we plan to write a codec procedure for DOB/0 right in `spore-sdk`, as it's written in TypeScript which is more familiar for the most of programmers in our eco-system.

# Related Issues
- [x] #107

### CODE REVIEW

**Comment by ashuralyk** [comment]
just ignore the failure of CI workflow, it caused by the conflict of `cross` and `cargo` version, which emits errors while compiling `spore-contracts`

File: packages/core/src/index.ts
Comment (Hanssen0) [inline packages/core/src/index.ts:1]: Do you think this is a planned modification?

File: packages/dob/src/codec/dob0/index.ts [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+    if (!moleculeLike.startsWith('0x')) {
+      moleculeLike = `0x${moleculeLike}`;
+    }
+    moleculeLike = bytes.bytify(moleculeLike);
+  }
+  const traitsBase = new DOB0.TraitsBase(moleculeLike.buffer);
+  return TraitsBase.unpack(traitsBase);
+}
+
+export function serailizePattern(traitsBase: TraitsBase, format?: 'hex'): ArrayBuffer | string {
```
Comment (Hanssen0) [inline packages/dob/src/codec/dob0/index.ts]: This interface design of multiple return types may cause some trouble with type checking. Considering this is the most essential exported function, I recommend clarifying the return type using function overloads.

**Review by Hanssen0 — CHANGES_REQUESTED** [review]

**Comment by Hanssen0** [comment]
btw, please also include the `.mol` file in the source code.

File: packages/core/src/index.ts
Comment (ashuralyk) [reply packages/core/src/index.ts:1]: sure, cobuild related interfaces should be exported

**Review by Hanssen0 — APPROVED** [review]
LGTM

**Review by Hanssen0 — APPROVED** [review]

**Comment by ashuralyk** [comment]
- [ ]

**Pull request closed without merging**